File: server/models/account.py
from typing import Optional

from pydantic import BaseModel, EmailStr, Field

from datetime import datetime


class ShippingAt(BaseModel):
    fname: str
    lname: str
    cname: str
    email: EmailStr
    town_city: str
    phone_number: str
    address: str
    country: str
    zip_code: int


class AccountSchema(BaseModel):
    customer_id: int = Field(...)
    account_id: int = Field(...)
    email: EmailStr = Field(...)
    password: str = Field(...)
    fname: str = Field(...)
    lname: str = Field(...)
    birth_date: str = Field(...)
    shipping_at: ShippingAt
    phone_area_code: str
    phone_number: str
    created: str
    active: bool = False
    active_date:  datetime
    get_noti: bool = False
    paypal: str
    fb_url: str
    registration_token: str
    expiration_token_date: datetime

    class Config:
        schema_extra = {
            "example": {
                "email": "abc@gmail.com",
                "password": "123456abc",
                "fname": "Hari",
                "lname": "Chan",
                "birth_date": "01/01/2001",
                "shipping_at": {
                    "fname": "A",
                    "lname": "Giang",
                    "cname": "Long",
                    "email": "a.giang@gmail.com",
                    "town_city": "Ho chi minh",
                    "phone_number": "+84 123456789",
                    "address": "53 barker street, Berminham, London",
                    "country": "England",
                    "zip_code": '35000'
                },
                "phone_area_code": "+84",
                "phone_number": "+84 123456789",
                "created": "",
                "active": False,
                "active_date":  None,
                "get_noti": False,
                "paypa": "",
                "fb_url": "",
                "registration_token": "",
                "expiration_token_date": "",
            }
        }


def ResponseModel(data, message):
    return {
        "data": [data],
        "code": 200,
        "message": message
    }


def ErrorResponseModel(error, code, message):
    return {"error": error, "code": code, "message": message}

File: databaseModel/account.js
var mongoose = require('mongoose');
var date = new Date();

var validateEmail = function (email) {
  var re = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
  return re.test(email);
};

var addressSchema = mongoose.Schema({
  address: {
    type: String,
  },
  city: {
    type: String,
  },
  zip_code: {
    type: String,
  },
});

var accountSchema = mongoose.Schema({
  email: {
    // required: true,
    type: String,
    lowercase: true,
    required: 'Email address is required',
    validate: [validateEmail, 'Please fill a valid email address'],
    match: [
      /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
      'Please fill a valid email address',
    ],
    // validate: {
    // 	validator: function(email){
    // 		var emailRegex = /^([\w-\.]+@([\w-]+\.)+[\w-]{2,4})?$/;
    // 		return emailRegex.test(email.text);
    // 	},
    // 	message: 'Email handle must have @'
    // }
  },
  password: {
    type: String,
    // required: true
  },
  fname: {
    type: String,
  },
  lname: {
    type: String,
  },
  birth_date: {
    type: Date,
  },
  shipping_at:
    // [addressSchema],
    [
      {
        fname: {
          default: null,
          type: String,
        },
        lname: {
          default: null,
          type: String,
        },
        cname: {
          default: null,
          type: String,
        },
        email: {
          default: null,
          type: String,
          lowercase: true,
        },
        town_city: {
          default: null,
          type: String,
        },
        phone_number: {
          default: null,
          type: String,
        },
        address: {
          type: String,
        },
        country: {
          type: String,
        },
        zip_code: {
          type: String,
        },
      },
    ],
  // {
  //     address:{
  //         type: String
  //     },
  //     city:{
  //         type: String

  //     },
  //     zip_code:{
  //         type: String
  //     },

  // },
  phone_area_code: {
    type: String,
    default: null,
  },
  phone_number: {
    default: null,
    type: String,
  },
  created: {
    type: Date,
    default: new Date(
      Date.UTC(
        date.getFullYear(),
        date.getMonth(),
        date.getDate(),
        date.getHours(),
        date.getMinutes(),
        date.getSeconds()
      )
    ),
  },
  active: {
    type: Boolean,
    default: false,
  },
  actived_date: {
    type: Date,
    default: null,
  },
  get_noti: {
    type: Boolean,
    default: false,
  },
  paypal: {
    type: String,
    default: null,
    lowercase: true,
    // required: 'Email address is required',
    // validate: [validateEmail, 'Please fill a valid email address'],
    // match: [/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/, 'Please fill a valid email address']
  },
  fb_url: {
    type: String,
    default: null,
    lowercase: true,
  },
  registration_token: {
    type: String,
    default: null,
  },
  expiration_token_date: {
    type: String,
    default: null,
  },
});
let Account = mongoose.model('Account', accountSchema);
module.exports = Account;
